### ISSUE TITLE
Material 3

### ISSUE THREAD

**Issue by yegor-pelykh** [darioielardi/flutter_speed_dial#279]
Hi!
Is it possible to make this control look like Material 3 FAB if that "Material 3" mode is used in the application?

Just for information: The "Material 3" theme mode can be set using the `useMaterial3` property of the `ThemeData` class when setting it as a theme in the `MaterialApp` widget.
![example](https://user-images.githubusercontent.com/21262076/195872146-80d5c160-844f-4345-baa4-307df70627ee.png)

In fact, I have already tried this "Material 3" mode in my app with a SpeedDial.
This control already looks pretty good when opened:
![speed-dial example](https://user-images.githubusercontent.com/21262076/195872667-e3abd630-f8f8-4578-8416-9cedce1c0212.png)

**But the main FAB button still looks wrong, which looks quite inappropriate.**

Could you fix this button with some styling, please?

For reference, the documentation for Material 3 FAB:
https://m3.material.io/components/floating-action-button

**Comment by nvi9** [comment]
You can specify the shape to override the currently default `StadiumBorder`, this way it looks similar to M3 FAB:

```dart
SpeedDial(
  shape: const RoundedRectangleBorder(
    borderRadius: BorderRadius.all(Radius.circular(16)),
  ),
  ...
)
```

**Comment by yegor-pelykh** [comment]
Yes, I can specify this shape, but in fact, I am now noticed that this is not enough to match Material 3. Both in part of the main fab, and in part of speed-dial children.
Could you please do a full inspection of the controls to match Material 3?

For the reference:
https://m3.material.io/components/floating-action-button/specs
Section "Small FAB layout".

![Specs](https://user-images.githubusercontent.com/21262076/196148733-fb48f653-d981-4730-8ec4-0e8916012f2c.png)

**Comment by nizioleque** [comment]
Hi,
after some tweaking I found the configuration that matches the M3 spec :)

Here is some example code

```dart
SpeedDial(
  shape: const RoundedRectangleBorder(
    borderRadius: BorderRadius.all(Radius.circular(16)),
  ),
  spacing: 16,
  childMargin: EdgeInsets.zero,
  childPadding: const EdgeInsets.all(8.0),
  children: [
    SpeedDialChild(
      shape: const RoundedRectangleBorder(
        borderRadius: BorderRadius.all(Radius.circular(12)),
      ),
    ),
    SpeedDialChild(
      shape: const RoundedRectangleBorder(
        borderRadius: BorderRadius.all(Radius.circular(12)),
      ),
    ),
  ]
)
```

and here is the result
![image](https://user-images.githubusercontent.com/92390086/210908674-5e9d5ccf-c636-494c-ac55-7424cbf96271.png)

and a comparison
![image](https://user-images.githubusercontent.com/92390086/210909092-a1390913-9e4c-43cc-82e2-aaeb41fdcfb2.png)
